src/sonic-yang-models/yang-models/sonic-static-route.yang

```diff
@@ -0,0 +1,124 @@
+    container STATIC_ROUTE {
+      description
+        "STATIC_ROUTE part of config_db.json";	
+	list STATIC_ROUTE_TEMPLATE_LIST {
+        key "prefix";
+        leaf prefix {
+          type inet:ip-prefix;
+          description
+            "prefix is the destination IP address, as key";
+        }
+        leaf nexthop {
+          type string;
+          description
+            "The next-hop that is to be used for the
+             static route as IP address. When interface needs to be
+             specified, use 0.0.0.0 as leaf value";
+        }
+        leaf ifname {
+          type string;
+          description
+            "When interface is specified, forwarding happens through it";
+        }
+        leaf advertise {
+          type string{
+            pattern "((true|false),)*(true|false)";
+                  }
+          default "false";
+       }
+      }
+      list STATIC_ROUTE_LIST {
+        key "vrf_name prefix";
+        leaf vrf_name {
+          type union {
+            type string {
+              pattern 'default';
+            }
+            type string {
+              pattern 'mgmt';
+            }
+            type string {
+              pattern "Vrf[a-zA-Z0-9_-]+";
+            }
+          }
+          description
+            "Virtual Routing Instance name as key";
+        }
+        leaf prefix {
+          type inet:ip-prefix;
+          description
+            "prefix is the destination IP address, as key";
+        }
+        leaf nexthop {
+          type string;
+          description
+            "The next-hop that is to be used for the
+             static route as IP address. When interface needs to be
+             specified, use 0.0.0.0 as leaf value";
+        }
+        leaf ifname {
+          type string;
+          description
+            "When interface is specified, forwarding happens through it";
+        }
+        leaf distance {
+          type string{
+	    pattern "((25[0-5]|2[0-4][0-9]|[0-1]?[0-9][0-9]?),)*(25[0-5]|2[0-4][0-9]|[0-1]?[0-9][0-9]?)";
+          }
+          default "0";
+          description
+            "Administrative Distance (preference) of the entry.  The
+             preference defines the order of selection when multiple
+             sources (protocols, static, etc.) contribute to the same
+             prefix entry.  The lower the preference, the more preferable the
+             prefix is.  When this value is not specified, the preference is
+             inherited from the default preference of the implementation for
+             static routes.";
+        }
+        leaf nexthop-vrf {
+          type string {
+            pattern "(((Vrf[a-zA-Z0-9_-]+)|(default)|(mgmt)),)*((Vrf[a-zA-Z0-9_-]+)|(default)|(mgmt))";
+          }
+          description
+            "VRF name of the nexthop. This is for vrf leaking";
+        }
+        leaf blackhole {
+          type string{
+            pattern "((true|false),)*(true|false)";

```

Comment:
How many boolean do you need?
Do you have any length limitation?

Reply:
These are comma separated values. We can use multiple comma separated values with the last value without comma. The regex mandates that pattern.

Reply:
Do you have max ECMP size? I don't think we can support unlimited length string.

Reply:
As part of https://github.com/sonic-net/sonic-buildimage/pull/11499 we decided we have custom validation subsequently as a new PR. Please take a look at the comments there. Thanks